src/test/run-make/type-mismatch-same-crate-name/Makefile

```diff
@@ -0,0 +1,19 @@
+-include ../tools.mk
+
+all:
+	# compile two different versions of crateA
+	$(RUSTC) --crate-type=rlib crateA.rs -C metadata=-1 -C extra-filename=-1
+	$(RUSTC) --crate-type=rlib crateA.rs -C metadata=-2 -C extra-filename=-2
+	# make crateB depend on version 1 of crateA
+	$(RUSTC) --crate-type=rlib crateB.rs --extern crateA=$(TMPDIR)/libcrateA-1.rlib
+	# make crateC depend on version 2 of crateA
+	$(RUSTC) crateC.rs --extern crateA=$(TMPDIR)/libcrateA-2.rlib 2>&1 | \
+		grep -z \
```

Comment:
The `aux-build` system is not powerful enough to set up the necessary crate conflicts. Basically, rustc needs to be passed `--extern crateA=...` for two different values of `crateA` when building two crates that end up linked together. (This actually happens when using cargo.)

I considered using `compile-flags` to set this up, but there is no way to get the necessary paths, since they depend on the current target platform and compilation stage. It ended up being easier just to write a makefile for this.

As for `tr -d '\r\n'` collapsing everything into a single line, this is mostly fine, since the line endings are not as important as the error being reported at all. (My regex for `grep -z` just skips over the intermediate parts anyway.) So if `tr -d '\r\n'` can be used on all platforms that makefile tests need to work on, I think this would be the best solution.